src/lib/nats/nats.ts

```diff
@@ -0,0 +1,7 @@
+import { NATS_URL } from "../../config";
+import { NatsConnection } from "nats";
+
+export function drain(natsConnection: NatsConnection): Promise<void> {
```

Comment:
Is there a way to specify that a parameter can/cannot be null/undefined/empty ?
It doesn't really make sense to call that function with a undefined parameter

Reply:
![image](https://user-images.githubusercontent.com/47562549/124926909-5541e080-dffe-11eb-8499-f50a11ec1a6e.png)

Is it what you meant ?

Reply:
there should always be a natsConnection, whether we're connected or not. If natsConnection is undefined then there's a bug in the code and as Thomas says it should be caught by the type system.
what's the reason for not calling natsConnection.drain() directly in the index?